scipy/stats/_discrete_distns.py

```diff
@@ -514,6 +526,144 @@ def _logcdf(self, k, M, n, N):
+    samples with :math:`k+r`'th sample being a failure. The former
+    can be modelled as a hypergeometric distribution. The probability
+    of the latter is simply the number of failures remaining
+    :math:`N-K-(r-1)` divided by the size of the remaining population
+    :math:`N-(k+r-1)`. This relationship can be shown as:
+
+    .. math:: NHG(k;N,K,r) = HG(k;N,K,k+r-1)\frac{(N-K-(r-1))}{(N-(k+r-1))}
+
+    where :math:`NHG` is probability mass function (PMF) of the
+    negative hypergeometric distribution and :math:`HG` is the
+    PMF of the hypergeometric distribution.
+
+    %(after_notes)s
+
+    Examples
+    --------
+    >>> from scipy.stats import nhypergeom
+    >>> import matplotlib.pyplot as plt
+
+    Suppose we have a collection of 20 animals, of which 7 are dogs.
+    Then if we want to know the probability of finding a given number
+    of dogs (successes) in a sample with exactly 12 animals that
+    aren't dogs (failures), we can initialize a frozen distribution
+    and plot the probability mass function:
+
+    >>> N, K, r = [20, 7, 12]
+    >>> rv = nhypergeom(N, K, r)
+    >>> x = np.arange(0, K+2)
+    >>> pmf_dogs = rv.pmf(x)
+
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111)
+    >>> ax.plot(x, pmf_dogs, 'bo')
+    >>> ax.vlines(x, 0, pmf_dogs, lw=2)
+    >>> ax.set_xlabel('# of dogs in our group with given 12 failures')
+    >>> ax.set_ylabel('nhypergeom PMF')
+    >>> plt.show()
+
+    Instead of using a frozen distribution we can also use `nhypergeom`
+    methods directly.  To for example obtain the probability mass
+    function, use:
+
+    >>> prb = nhypergeom.pmf(x, N, K, r)
+
+    And to generate random numbers:
+
+    >>> R = nhypergeom.rvs(N, K, r, size=10)
+
+    To verify the relationship between `hypergeom` and `nhypergeom`, use:
+
+    >>> from scipy.stats import hypergeom, nhypergeom
+    >>> N, K, r = 45, 13, 8
+    >>> k = 6
+    >>> nhypergeom.pmf(k, N, K, r)
+    0.06180776620271643
+    >>> hypergeom.pmf(k, N, K, k+r-1) * (N - K - (r-1)) / (N - (k+r-1))
+    0.06180776620271644
+
+    See Also
+    --------
+    hypergeom, binom, nbinom
+
+    References
+    ----------
+    .. [1] Negative Hypergeometric Distribution on Wikipedia
+           https://en.wikipedia.org/wiki/Negative_hypergeometric_distribution
+
+    .. [2] Negative Hypergeometric Distribution from
+           http://www.math.wm.edu/~leemis/chart/UDR/PDFs/Negativehypergeometric.pdf
+
+    """
+    def _get_support(self, N, K, r):
+        return 0, K
+
+    def _argcheck(self, N, K, r):
+        cond = (K >= 0) & (K <= N)
+        cond &= (r >= 0) & (r <= N-K)
+        return cond
+
+    def _logpmf(self, k, N, K, r):
+        cond = ((r == 0) & (k == 0))
+        result = (-betaln(k+1, r) + betaln(k+r, 1) -
+                  betaln(K-k+1, N-r-K+1) + betaln(N-r-k+1, 1) +
+                  betaln(K+1, N-K+1) - betaln(N+1, 1))
+        result[cond] = 0.0
```

Comment:
To avoid the warning that occurs when `r` is 0, this can be rewritten using `_lazywhere`:
```suggestion
        result = _lazywhere(~cond, (k, N, K, r),
                            lambda k, N, K, r:
                                (-betaln(k+1, r) + betaln(k+r, 1) -
                                 betaln(K-k+1, N-r-K+1) + betaln(N-r-k+1, 1) +
                                 betaln(K+1, N-K+1) - betaln(N+1, 1)),
                            fillvalue=0.0)
```

Reply:
Yeah! Thanks. This was very helpful. It now works without any warnings.